### DISCUSSION TITLE
Bug Report "Double Where"

### DISCUSSION THREAD

**Discussion by asapostolov** [ravendb/ravendb#18969] [General]
I started seeing this in RavenDB 6.0.105 local Client and Server. I have a query that looks like this:
``` CSharp
var portalUsersAll = RavenSession.Query<PortalUser>()
                              .Where( x => x.Id.In( portalUserIds ) )
                              .Where( x => x.IsActive == true && x.IsArchived == false )
                              .ToList();
```
This query returns 0 Records when I input 1 PortalUser Id
``` CSharp
var portalUsersAll = RavenSession.Query<PortalUser>()
                              .Where( x => x.Id.In( portalUserIds ) 
                              &&  x.IsActive == true && x.IsArchived == false )
                              .ToList();
```
The second one returns a single record (correctly) with the same PortalUser Id

The "Double Where" worked for a lot of versions, I'm not sure why it's not working here. The Cloud build 6.0.104-cloud-60036 (my live server) doesn't have this issue.

**Comment by Lwiel** [comment]
I reproduced the issue on v6.0.105 using Corax, are you using this search engine as well?

On development branch (https://github.com/ravendb/ravendb/tree/v6.0) it's already fixed, I get the correct number of results.

**Comment by asapostolov** [comment]
I actually haven’t specified any engine, but I found that two of my indexes
are breaking because they moved to Corax. Is Corax going to be the default
engine from now on?

It’s good that it’s already fixed!

**Reply by arekpalinski** [reply]
Corax is the default indexing engine for Dev and Community licenses only. It applies to newly created indexes (or if they are reset), existing ones stay on Lucene (if you upgrade RavenDB server from version 5.4 to 6.0 they won't automatically switch to Corax).

Professional and Enterprise licenses still default to Lucene in RavenDB 6.0.

**Comment by Lwiel** [comment]
I added this particular case to our tests, here's the PR merged into v6.0 - https://github.com/ravendb/ravendb/pull/18987
